statsmodels/discrete/discrete_model.py

```diff
@@ -2588,17 +2594,20 @@ def _hessian_nb1(self, params):
 
         a1 = mu/alpha
 
+        prob = a1 / (a1+mu)
+        # Note: 1/(alpha+1) == (mu/alpha) / (mu + mu/alpha) == a1/(mu+a1)
```

Comment:
the main question here is numerical stability in the neighborhood of alpha=0
```
>>> mu=1e1; alpha=np.float64(0); (mu/alpha) / (mu + mu/alpha) - 1, 1/(alpha+1) - 1
(nan, 0.0)
```
#3863 i.e. we don't support alpha=0 currently, but this goes in the wrong direction for improving this.

Reply:
a1 = mu/alpha is defined before we get to this point.  I'm not indifferent to this concern, but it's orthogonal to this PR.

Reply:
I know that at the moment we cannot avoid a1 = mu/alpha
but the part 1/(alpha+1) is well defined at alpha=0, i.e. it is worth creating a tmp variable for it directly if it is repeatedly used, instead of going through a1.

Reply:
We can cross that road when we get to it.  This is just simplifying algebra (to avoid repeated calls).  Unrelated to design decisions.

Reply:
> This is just simplifying algebra

No, it's not. You change the numerical properties of the estimator. And AFAICS, in this case there is no reason to do so, by using a second temp variable

Reply:
> You change the numerical properties of the estimator

Ignoring the fact that you are worrying about a case that isn't supported, I claim that if you plug in the unsupported parameter `alpha=0`, you get nans or infs both in master and in the PR.  Both sets of expressions involve division by zero, but in the PR there is division by zero in _fewer places_, which will make them easier to address when we eventually do care about this case.

> And AFAICS, in this case there is no reason to do so, by using a second temp variable

Tell that to whoever wrote `NegativeBinomial._ll_nbin`.

Clarifying the relationship between the nb1 and nb2 cases is worthwile.

Would your concerns be sufficiently addressed by an additional check:

```
if alpha == 0:
    prob = 1
```
?